### ISSUE TITLE
Switching networks causes filter/event callbacks to silently fail

### ISSUE THREAD

**Issue by iwooden** [MetaMask/metamask-extension#2252]
When I first start up Chrome and use Metamask, web3 fires filter/event callbacks as expected. This works for the Ethereum mainnet, as well as the Rinkeby and Kovan test networks.

However, after changing networks in Metamask, filters/event callback functions silently stop being called. Once the network has been switched, any site the user visits will be unable to use filters/event callbacks. This behavior remains until Chrome is restarted.

**Metamask version**: 3.10.8

**Repro steps**: Start a fresh Chrome session. Paste following code into the Chrome javascript console:
```
var lastestBlockFilter = web3.eth.filter('latest');
lastestBlockFilter.watch((err, res) => {
    console.log(res);
});
```

You should get a stream of block hashes in the console. Switch networks in Metamask, refresh, and paste the code into the console again. Block information will stop being populated. This persists even if you switch back to the original network.

Probably related to the following issue:
#2114

**Comment by lazaridiscom** [comment]
@iwooden , please try this fix (i tried your test-case, and it works)

https://github.com/MetaMask/metamask-extension/pull/3224/files

**Comment by lazaridiscom** [comment]
@Zanibas , this one was fixed with PR #3224
